### PR TITLE
Replace withColumn with select for columnar dataframe

### PR DESCRIPTION
**Pull request by lijinf2**
When withColumn is used inside a for loop to add new columns iteratively, it gets slower as the number of columns in the dataframe increases.

### CODE REVIEW

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/core.py [hunk 1]
```diff
@@ -1306,6 +1304,12 @@ def _pre_process_data(
                     raise ValueError(
                         "All columns must be integral types or float/double types."
                     )
+
+            taglen = len("_c3BhcmstcmFwaWRzLW1sCg==")
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/core.py:1308]: Better to define the tag in a new variable so all references to it are consistent.

**Comment by lijinf2** [comment]
build

**Review by eordentlich — DISMISSED** [review]
👍

**Comment by lijinf2** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**